### PR TITLE
32 improve documentation with JavaDoc

### PR DESCRIPTION
**Pull request by ARoy667**
This is an ongoing PR which I will continue to push to as we add more files to our codebase. So far I have added JavaDoc for virtually every public interface and their implementing classes/methods, as well as for constructors for every class that has been pushed to Main. 
At the same time, I have also been handling Issue 30 (refactor code for consistent formatting) by trying to catch any discrepancies and following a consistent solution for spacing, empty lines, spacing between braces, etc.

Feel free to suggest any edits for the descriptions I added or if I missed anything :)

### CODE REVIEW

File: src/entity/CommonGame.java [hunk 1]
```diff
@@ -103,9 +187,19 @@ public boolean isGameOver() {
...
+     * Adds one to current score.
+     */
     @Override
-    public void incrementScore() { this.score += 1; }
+    public void incrementScore() {
+        this.score += 1;
+    }
 
+    /**
+     * Subtracts one from current score.
```
Comment (rayceramsay) [inline src/entity/CommonGame.java:199]:
```suggestion
     * Subtracts one from current lives.
```

File: src/view/LoadableGamesViewFactory.java [hunk 2]
```diff
@@ -23,6 +33,14 @@ public static LoadableGamesView create(ViewManagerModel viewManagerModel,
         return new LoadableGamesView(viewManagerModel, getLoadableGamesViewModel, loadGameController);
     }
 
+    /**
+     * Creates instance pf LoadGameController
```
Comment (rayceramsay) [inline src/view/LoadableGamesViewFactory.java:37]:
```suggestion
     * Creates an instance of LoadGameController
```

**Comment by rayceramsay** [comment]
Wow, this is awesome Aabha! Thanks for doing this!

Just a friendly reminder to merge main into this branch every so often so that there aren't big conflicts from things like documenting files that got deleted/significantly changed or stuff like that.

**Review by ethanmcf — APPROVED** [review]
Damn this a lot of comments, good job!

**Review by NickinAction — APPROVED** [review]
Looks good!

**Review by ethanmcf — CHANGES_REQUESTED** [review]
Just want to remove the keys from api.env before we merge to main

**Comment by ARoy667** [comment]
I saw that Rayce has added JavaDoc comments to his existing SQLite database PR so if that is the only thing left to merge into main I think we should be good to merge this as well

**Review by Artakarak — APPROVED** [review]
Just as we needed.

**Pull request merged**